crates/compiler/src/output/declaration.rs

```diff
@@ -121,15 +119,25 @@ impl Declaration {
         self
     }
 
-    pub fn with_type(mut self, r#type: impl Into<Type>) -> Self {
-        self.r#type = r#type.into();
-        self
-    }
-
     pub fn with_range(mut self, range: impl Into<RangeInclusive<Position>>) -> Self {
         self.range = Some(range.into());
         self
     }
+
+    pub fn eq(&self, other: &Self, epsilon: f32) -> Result<bool, InvalidCastError> {
```

Comment:
Hmm. Feels unfortunate to not implement the `PartialEq` trait instead, but I guess it's not possible here due to the cast error.

Reply:
I think I might be able to refactor it into the right trait later, let me see

Reply:
Update: We _do_ implement `PartialEq` as well, but this method is more suitable for the internal float comparisons.